assets/js/gutenberg-plugin.js

```diff
@@ -179,11 +179,16 @@ const DistributorPlugin = () => {
 		return null;
 	}
 
+	const distributorTopMenu = document.querySelector(
+		'#wp-admin-bar-distributor'
+	);
+
 	// If we are on a non-supported post status, change what we show
 	if (
 		dtGutenberg.supportedPostStati &&
 		! dtGutenberg.supportedPostStati.includes( postStatus )
 	) {
+		distributorTopMenu.classList.add( 'hide' );
```

Comment:
Untested but this should avoid errors in the JS if the element is not defined for whatever reason.

```suggestion
		distributorTopMenu && distributorTopMenu.classList.add( 'hide' );
```

Reply:
eslint doesn't like this syntax, so I used the optional chaining operator.

Reply:
> eslint doesn't like this syntax, so I used the optional chaining operator.

NP, works for me.